firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -322,42 +338,48 @@ public void signOutTester() {
-    if (activity instanceof SignInResultActivity) {
-      // SignInResult is internal to the SDK and is destroyed after creation
-      return;
-    }
-    if (activity.isChangingConfigurations()) {
-      remakeSignInConfirmationDialog =
-          signInConfirmationDialog != null && signInConfirmationDialog.isShowing();
-      remakeUpdateDialog = updateDialog != null && updateDialog.isShowing();
-      dismissDialogs();
-      return;
+  void onActivityResumed(Activity activity) {
+    if (awaitingSignInDialogConfirmation()) {
+      if (dialogHostActivity != null && dialogHostActivity != activity) {
+        showSignInDialogTask.setException(
+            new FirebaseAppDistributionException(
+                ErrorMessages.HOST_ACTIVITY_INTERRUPTED, HOST_ACTIVITY_INTERRUPTED));
+      } else {
+        showSignInConfirmationDialog(activity);
+      }
     }
 
-    if (signInConfirmationDialog != null && signInConfirmationDialog.isShowing()) {
-      setCachedUpdateIfNewReleaseCompletionError(
-          new FirebaseAppDistributionException(
-              ErrorMessages.AUTHENTICATION_CANCELED, AUTHENTICATION_CANCELED));
+    if (awaitingUpdateDialogConfirmation()) {
+      if (dialogHostActivity != null && dialogHostActivity != activity) {
+        showUpdateDialogTask.setException(
+            new FirebaseAppDistributionException(
+                ErrorMessages.HOST_ACTIVITY_INTERRUPTED, HOST_ACTIVITY_INTERRUPTED));
+      } else {
+        synchronized (cachedNewReleaseLock) {
+          showUpdateConfirmationDialog(
+              activity, ReleaseUtils.convertToAppDistributionRelease(cachedNewRelease));
+        }
+      }
     }
+  }
 
-    if (updateDialog != null && updateDialog.isShowing()) {
-      setCachedUpdateIfNewReleaseCompletionError(
-          new FirebaseAppDistributionException(
-              ErrorMessages.UPDATE_CANCELED, Status.INSTALLATION_CANCELED));
+  @VisibleForTesting
+  void onActivityPaused(Activity activity) {
+    if (activity == dialogHostActivity) {
+      synchronized (updateIfNewReleaseTaskLock) {
+        remakeSignInConfirmationDialog =
+            signInConfirmationDialog != null && signInConfirmationDialog.isShowing();
+        remakeUpdateConfirmationDialog =
+            updateConfirmationDialog != null && updateConfirmationDialog.isShowing();
+      }
+      dismissDialogs();
     }
   }
 
-  void onActivityCreated(Activity activity) {
-    if (remakeSignInConfirmationDialog) {
-      remakeSignInConfirmationDialog = false;
-      showSignInConfirmationDialog(activity);
-    } else if (remakeUpdateDialog) {
-      remakeUpdateDialog = false;
-      synchronized (cachedNewReleaseLock) {
-        showUpdateAlertDialog(
-            activity, ReleaseUtils.convertToAppDistributionRelease(cachedNewRelease));
-      }
+  @VisibleForTesting
+  void onActivityDestroyed(@NonNull Activity activity) {
+    if (activity == dialogHostActivity) {
+      dialogHostActivity = null;
```

Comment:
On second thought, I think we should add a comment here that this, in part, makes the configuration change case work, since it signals that in onResume we should not bother checking the activity reference.

Reply:
Will add